src/archive_librarian/dbHandler.cpp

```diff
@@ -0,0 +1,1456 @@
+    if (sqlite3_prepare_v2(db_, jobLookupSQL, -1, &jobSelectStmt_, nullptr) != SQLITE_OK) {
+        printf("[ERROR] Failed to prepare jobSelectStmt: %s\n", sqlite3_errmsg(db_));
+        jobSelectStmt_ = nullptr;
+    }
+
+}
+
+DBHandler::~DBHandler() {
+    if (jobIdLookupStmt_) {
+        sqlite3_finalize(jobIdLookupStmt_);
+        jobIdLookupStmt_ = nullptr;
+    }
+
+    if (userInsertStmt_) {
+        sqlite3_finalize(userInsertStmt_);
+        userInsertStmt_ = nullptr;
+    }   
+
+    if (userSelectStmt_) {
+        sqlite3_finalize(userSelectStmt_);
+        userSelectStmt_ = nullptr;
+    }
+
+    if (jobListInsertStmt_) {
+        sqlite3_finalize(jobListInsertStmt_);
+        jobListInsertStmt_ = nullptr;
+    }
+
+    if (jobListSelectStmt_) {
+        sqlite3_finalize(jobListSelectStmt_);
+        jobListSelectStmt_ = nullptr;
+    }
+
+    if (jobInsertStmt_) {
+        sqlite3_finalize(jobInsertStmt_);
+        jobInsertStmt_ = nullptr;
+    }
+
+    if (jobSelectStmt_) {
+        sqlite3_finalize(jobSelectStmt_);
+        jobSelectStmt_ = nullptr;
+    }
+
+    
+
+    if (db_) {
+        int rc = sqlite3_close(db_);
+        if (rc != SQLITE_OK) {
+            printf("[ERROR] Failed to close db: %s \n", sqlite3_errmsg(db_));
+        }
+        db_ = nullptr;
+    }
+}
+
+bool DBHandler::initializeFromSchema(const std::string& schemaPath) {
+    std::ifstream file(schemaPath);
+    if (!file.is_open()) {
+        printf("[ERROR] Failed to open schema file: %s\n", schemaPath.c_str());
+        return false;
+    }
+
+    std::stringstream buffer;
+    buffer << file.rdbuf();
+    std::string schemaSQL = buffer.str();
+
+    // Enable foreign key constraints
+    char* errMsg = nullptr;
+    int rc = sqlite3_exec(db_, "PRAGMA foreign_keys = ON;", nullptr, nullptr, &errMsg);
+    if (rc != SQLITE_OK) {
+        printf("[ERROR] Failed to enable foreign keys: %s\n", errMsg);
+        sqlite3_free(errMsg);
+        return false;
+    }
+
+    rc = sqlite3_exec(db_, schemaSQL.c_str(), nullptr, nullptr, &errMsg);
+    if (rc != SQLITE_OK) {
+        printf("[ERROR] Failed to initialize DB schema: %s\n", errMsg);
+        sqlite3_free(errMsg);
+        return false;
+    }
+
+    return true;
+}
+
+bool DBHandler::clearCache() const {
+    jobIdCache_->clear();
+    size_t currentSize = jobIdCache_->size();
+    return(currentSize == 0);
+}
+
+// For Librarian to check whether DBHandler has been initialized and works properly upon startup
+bool DBHandler::testConnection() {
+    // 1. Check database handle exists
+    if (!db_) {
+        fprintf(stderr, "[DBHandler] Database handle is null\n");
+        return false;
+    }
+    
+    // 2. Test basic connectivity
+    int result = sqlite3_exec(db_, "SELECT 1;", nullptr, nullptr, nullptr);
+    if (result != SQLITE_OK) {
+        fprintf(stderr, "[DBHandler] Basic query failed: %s\n", sqlite3_errmsg(db_));
+        return false;
+    }
+    
+    // 3. Verify required tables exist
```

Comment:
I like the verification that the database has the expected tables, however this test does not seem like it has anything to do with wether or not we can actually connect to the database. We should do one of the following:

1. Rename this function from `testConnection` to something else
2. Separate the third check to a secondary function called something like `verifyTables`.